### PR TITLE
Include che.devworkspaces.enabled property

### PR DESCRIPTION
**Pull request by xbaran4**
### What does this PR do?
Adds property che.devworkspaces.enabled to che.properties. Exposes it with /api/workspace/settings endpoint.

### What issues does this PR fix or reference?

#19160

### How to test this PR?

Build and deploy on minikube.
Query /api/workspace/settings endpoint.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 1]
```diff
@@ -18,18 +18,7 @@
...
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_DEVFILE_REGISTRY_INTERNAL_URL_PROPERTY;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_DEVFILE_REGISTRY_URL_PROPERTY;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_PLUGIN_REGISTRY_INTERNAL_URL_PROPERTY;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_PLUGIN_REGISTRY_URL_PROPERTY;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_STORAGE_AVAILABLE_TYPES;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_STORAGE_PREFERRED_TYPE;
-import static org.eclipse.che.api.workspace.shared.Constants.DEBUG_WORKSPACE_START;
-import static org.eclipse.che.api.workspace.shared.Constants.DEBUG_WORKSPACE_START_LOG_LIMIT_BYTES;
-import static org.eclipse.che.api.workspace.shared.Constants.WORKSPACE_INFRASTRUCTURE_NAMESPACE_ATTRIBUTE;
+import static org.eclipse.che.api.workspace.shared.Constants.*;
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java]: Revert this change, please.

File: wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceServiceTest.java [hunk 2]
```diff
@@ -23,14 +23,7 @@
...
 import static org.eclipse.che.api.workspace.server.DtoConverter.asDto;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_FACTORY_DEFAULT_EDITOR_PROPERTY;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_FACTORY_DEFAULT_PLUGINS_PROPERTY;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_AUTO_START;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_STORAGE_AVAILABLE_TYPES;
-import static org.eclipse.che.api.workspace.shared.Constants.CHE_WORKSPACE_STORAGE_PREFERRED_TYPE;
-import static org.eclipse.che.api.workspace.shared.Constants.DEBUG_WORKSPACE_START;
-import static org.eclipse.che.api.workspace.shared.Constants.DEBUG_WORKSPACE_START_LOG_LIMIT_BYTES;
-import static org.eclipse.che.api.workspace.shared.Constants.SUPPORTED_RECIPE_TYPES;
+import static org.eclipse.che.api.workspace.shared.Constants.*;
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceServiceTest.java]: revert this change, please

**Review by skabashnyuk — COMMENTED** [review]
In general looks good. I think you can `undraft` this pr.

**Comment by sleshchenko** [comment]
Sorry or thank me for undrafting this PR, I did it to get che-server image built by CI, which we can use for testing integration with dashboard.

**Review by sparkoo — COMMENTED** [review]
lgtm except few nitpicks. Also you don't have all commits signed, so you would need to fix that to make this PR mergeable.

**Review by mshaposhnik — APPROVED** [review]
LGTM along with requested changes fixed

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties [hunk 3]
```diff
@@ -158,6 +158,9 @@ che.workspace.startup_debug_log_limit_bytes=10485760
 # This configuration is mainly required for workspace idling when the OpenShift OAuth is enabled.
 che.workspace.stop.role.enabled=true
 
+#Specifies whether che is deployed with devworkspaces enabled.
```
[the remark this answers is not in the document]
Comment (metlos) [reply assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties]: I would possibly add more detail to this, so that the users don't expect devworkspace support to magically appear by just flipping this to "true".

Something along the lines of:

```
This property is set by the Che operator if it also installed the support for DevWorkspaces. This property is used
to advertise this fact to the Che dashboard. It does not make sense to change the value of this property manually.
```

**Review by skabashnyuk — APPROVED** [review]

**Review by metlos — APPROVED** [review]

**Comment by xbaran4** [comment]
quay.io/pbaran/che-server:latest

**Comment by skabashnyuk** [comment]
Tested on `quay.io/skabashn/che-server:devworkspace`
Знімок екрана 2021-03-04 о 12 37 55
